libutil/utils.c

```diff
@@ -350,49 +350,11 @@ reboot_system()
 
 #    endif /* WINDOWS */
 
-/* this sucks.
- * i can't believe this is best way to implement this in Win32...
- *  but i can't seem to find a better way.
- * msdn suggests using CreateFile() with CREATE_NEW or OPEN_EXISTING,
- *   and then checking error codes; but the problem there is that C:\\
- *   returns PATH_NOT_FOUND regardless. */
 bool
 file_exists(const TCHAR *fn)
 {
 #    ifdef WINDOWS
-    WIN32_FIND_DATA fd;
-    HANDLE search;
-
-    DO_ASSERT(fn != NULL);
-
-    search = FindFirstFile(fn, &fd);
-
-    if (search == INVALID_HANDLE_VALUE) {
-
-        /* special handling for e.g. C:\\ */
-        if (LAST_WCHAR(fn) == L'\\' || LAST_WCHAR(fn) == L':') {
-            WCHAR buf[MAX_PATH];
-            _snwprintf(buf, MAX_PATH, L"%S%S*", fn,
-                       LAST_WCHAR(fn) == L'\\' ? L"" : L"\\");
-            NULL_TERMINATE_BUFFER(buf);
-            search = FindFirstFile(buf, &fd);
-            if (search != INVALID_HANDLE_VALUE) {
-                FindClose(search);
-                return TRUE;
-            } else {
-                DO_DEBUG(
-                    DL_VERB,
-                    printf("%S: even though we tried hard, %d\n", buf, GetLastError()););
-            }
-        }
-
-        DO_DEBUG(DL_VERB,
-                 printf("%S doesn't exist because of: %d\n", fn, GetLastError()););
-        return FALSE;
-    } else {
-        FindClose(search);
-        return TRUE;
-    }
+    return GetFileAttributes(fn) != INVALID_FILE_ATTRIBUTES;
```

Comment:
One concern is whether this works on all platforms.  MSDN for `GetFileAttributes` talks about it only being available on XP+ which may be why it was not used here (this code had to run on NT and 2K originally).  We no longer officially support NT or 2K in auxiliary code, like this here in libutil/, and don't test on NT or 2K for core/, though if it's easy to not consciously break 2K in core/ we'd prefer to not break it.  However, `NtQueryAttributesFile` certainly existed on NT: so maybe `GetFileAttributes` did as well and the MSDN pages these days don't bother to say whether something works on NT or 2K?

Reply:
[FindFirstFile](https://docs.microsoft.com/en-us/windows/desktop/api/fileapi/nf-fileapi-findfirstfilea) from the original commit also marked by MSDN as the minimum version is Windows XP... so yes, I suppose they just don't care of NT anymore.